src/routes/safe/components/Apps/components/AppFrame.tsx

```diff
@@ -114,7 +118,17 @@ const AppFrame = ({ appUrl }: Props): ReactElement => {
       clearTimeout(errorTimer.current)
     }
 
-    if (appIsLoading) {
+    if (appIsLoading && currentApp) {
+      const trackData = loadFromStorage<AppTrackData>(APPS_DASHBOARD) || {}
+      let currentOpenCount = trackData[currentApp.id]?.openCount
+      saveToStorage(APPS_DASHBOARD, {
+        ...trackData,
+        [currentApp.id]: {
+          ...trackData[currentApp.id],
+          timestamp: Date.now(),
+          openCount: currentOpenCount ? ++currentOpenCount : 1,
+        },
+      })
```

Comment:
I would suggest you abstract this logic into a tracking function and then two higher level functions.